content/sensu-go/6.1/operations/maintain-sensu/troubleshoot.md

```diff
@@ -379,6 +379,123 @@ Since the `lsb_release` package is not installed, the agent will not be able to
 
+## Investigate etcd cluster status
+
+Some issues require you to investigate the state of the etcd cluster or data stored within etcd.
+In these cases, we suggest using the `etcdctl` tool to query and manage the etcd database.
+
+Sensu's supported packages do not include the `etcdctl` executable, so you must get it from a compatible etcd release.
+
+### Configure etcdctl environment variables
+
+To use etcdctl to investigate etcd cluster and data storage issues, first run these commands to configure etcdctl environment variables:
+
+{{< code shell >}}
+export ETCDCTL_API=3
+export ETCDCTL_CACERT=/etc/sensu/ca.pem
+# skip ETCDCTL_CERT and ETCDCTL_KEY, unless your etcd uses client certificate authentication
+# export ETCDCTL_CERT=/etc/sensu/cert.pem
+# export ETCDCTL_KEY=/etc/sensu/key.pem
+export ETCDCTL_ENDPOINTS="https://backend01:2379,https://backend02:2379,https://backend03:2379"
+{{< /code >}}
+
+### View cluster status and alarms
+
+Use the commands listed here to retrieve etcd cluster status or list alarms.
+{{< code shell >}}
+# get status
+etcdctl endpoint status
+# list alarms
+etcdctl alarm list
+# clear alarms
+etcdctl alarm dearm
+{{< /code >}}
+
+### Restore a cluster with an oversized database 
+
+The etcd default maximum database size is 2 GB.
+If you suspect your etcd database has exceeded the maximum size, run this command to confirm cluster size:
+
+{{< code shell >}}
+# get current status with database size
+etcdctl endpoint status
+https://backend01:2379, 88db026f7feb72b4, 3.3.22, 2.1GB, false, 144, 18619245
+https://backend02:2379, e98ad7a888d16bd6, 3.3.22, 2.1GB, true, 144, 18619245
+https://backend03:2379, bc4e39432cbb36d, 3.3.22, 2.1GB, false, 144, 18619245
+{{< /code >}}
+
+To restore an etcd cluster with a database size that exceeds 2 GB, run these commands:
+
+{{< code shell >}}
+# get current revision number
+etcdctl endpoint status --write-out="json" | egrep -o '"revision":[0-9]*' | egrep -o '[0-9].*'
+
+# compact to revision, substitute revision obtained above for $rev
+etcdctl compact $rev
+
+# defrag to free space
+etcdctl defrag
+
+# confirm effective
+etcdctl endpoint status
+https://backend01:2379, 88db026f7feb72b4, 3.3.22, 1.0 MB, false, 144, 18619245
+https://backend02:2379, e98ad7a888d16bd6, 3.3.22, 1.0 MB, true, 144, 18619245
+https://backend03:2379, bc4e39432cbb36d, 3.3.22, 1.0 MB, false, 144, 18619245
+{{< /code >}}
+
+## Datastore performance
+
+### Overview
+
+In a default deployment, Sensu uses [etcd datastore][17] for both configuration and state. As the number of checks and entities in your Sensu installation increases, so does the volume of read and write requests to etcd database.
+
+One trade-off in etcd's design is it's sensitivity to disk and CPU latency. When certain latency tollerances are regularly exceeded, failures will cascade. Sensu will attempt to recover from these conditions when it can, but 
```

Comment:
```suggestion
One trade-off in etcd's design is its sensitivity to disk and CPU latency.
When certain latency tolerances are regularly exceeded, failures will cascade.
Sensu will attempt to recover from these conditions when it can, but 
```

Reply:
Something is missing here after the "but"